torch/_dynamo/bytecode_transformation.py

```diff
@@ -397,7 +397,12 @@ def fix_vars(instructions: List[Instruction], code_options):
         if instructions[i].opcode in HAS_LOCAL:
             instructions[i].arg = varnames[instructions[i].argval]
         elif instructions[i].opcode in HAS_NAME:
-            instructions[i].arg = names[instructions[i].argval]
+            if not instructions[i].argval in names:
```

Comment:
Other places in the code add names to `code_options["co_names"]`. We should either assert for existence here; or only add global names here and not elsewhere.

Reply:
I think this may be an unnecessary artifact from a earlier separate attempt

Reply:
reverted and added to code_options on per-needed basis.